src/PowerPing/Program.cs

```diff
@@ -555,34 +555,29 @@ static void Main(string[] args)
                     PowerPing.Lookup.QueryWhoIs(attributes.Host);
                     break;
                 case "graphing":
-                    g = new Graph(attributes.Host);
+                    g = new Graph(attributes.Host, cancellationTokenSource.Token);
                     g.Start();
                     break;
                 case "compactgraph":
-                    g = new Graph(attributes.Host);
+                    g = new Graph(attributes.Host, cancellationTokenSource.Token);
                     g.CompactGraph = true;
                     g.Start();
                     break;
                 case "flooding":
-                    thread = new Thread(() =>
-                    {
-                        p.Flood(attributes.Host);
-                    });
-                    thread.Start();
+                    p.Flood(attributes.Host);
                     break;
                 case "scanning":
                     p.Scan(args.Last());
                     break;
                 default:
                     // Send ping normally
-                    thread = new Thread(() =>
-                    {
-                        p.Send(attributes);
-                    });
-                    thread.Start();
+                    p.Send(attributes);
                     break;
             }
-           
+
+            // Reset console colour
+            Display.ResetColor();
+            Console.CursorVisible = true;
```

Comment:
Used to be in the exit handler. Now the calls inside the case statement are blocking, and once the cancellation token is set those calls will finish gracefully and proceed to run this bit.

Reply:
Excellent 👍